ios/Breez Notification Service Extension/BreezManager.swift

```diff
@@ -11,55 +11,66 @@ fileprivate var log = Logger(
 fileprivate var log = Logger(OSLog.disabled)
 #endif
 
-// SDK events listener
-class SDKListener: EventListener {
+class BreezManager {
+    private static var breezSDK: BlockingBreezServices? = nil
+    fileprivate static var queue = DispatchQueue(label: "BreezManager")
+    fileprivate static var sdkListener: EventListener? = nil
     
-    var paymentListener : PaymentListener
+    static func register(listener: EventListener) throws -> BlockingBreezServices? {
+        try BreezManager.queue.sync { [] in
+            BreezManager.sdkListener = listener
+            if BreezManager.breezSDK == nil {
+                BreezManager.breezSDK = try BreezManager.connectSDK()
+            }
+            return BreezManager.breezSDK
+        }
+    }
     
-    init(paymentListener: @escaping PaymentListener) {
-        self.paymentListener = paymentListener
+    static func unregister() {
+        BreezManager.queue.sync { [] in
+            BreezManager.sdkListener = nil            
+        }
     }
     
-    func onEvent(e: BreezEvent) {
-        switch e {
-        case .invoicePaid(details: let details):
-            log.info("Received payment. Bolt11: \(details.bolt11)\nPayment Hash:\(details.paymentHash)")
-            if details.payment != nil {
-                paymentListener(details.payment!)
-            }
-            return
-        default:
-            break
+    static func connectSDK() throws -> BlockingBreezServices? {
+        log.trace("connectSDK()")
+        try setLogStream(logStream: SDKLogListener(logger: NotificationService.logger))
+        
+        // Create the default config
+        let apiKey = try Environment.glApiKey()
+        log.trace("API_KEY: .\(apiKey)")
+        var config = defaultConfig(envType: EnvironmentType.production, apiKey: apiKey,
+                                   nodeConfig: NodeConfig.greenlight(
+                                    config: GreenlightNodeConfig(partnerCredentials: nil, inviteCode: nil)))
+        
+        config.workingDir = FileManager
+            .default.containerURL(forSecurityApplicationGroupIdentifier: "group.F7R2LZH3W5.com.cBreez.client")!
+            .absoluteString
+        
+        // Construct the seed
+        let mnemonic = CredentialsManager.shared.restoreMnemonic() ?? ""
+        log.trace("mnemonic: .\(mnemonic)")
+        let seed = try? mnemonicToSeed(phrase: mnemonic)
+        // Connect to the Breez SDK make it ready for use
+        guard seed != nil else {
+            throw SdkError.Generic(message: "seed not found")
         }
```

Comment:
This can be reduced into a single guard statement and in light of converting mnemonic into a guard statement, we can use a better representative error message here.
```suggestion
        guard let seed = try? mnemonicToSeed(phrase: mnemonic) else {
            throw SdkError.Generic(message: "failed to generate seed")
        }
        // Connect to the Breez SDK make it ready for use
```
and the comment line has to be moved below.

Reply:
Done